docs/proposals/spec-reconciliation.md

```diff
@@ -0,0 +1,91 @@
+# Spec Reconciliation Proposal
+
+## Introduction
+Add a way for broker to sync the list of specs it has with registries while handling the following corner cases:
+ 1. Temporary loss of connection with registry
+ 2. A spec removed from the registry but has provisioned instance/s
+
+## Problem Description
+
+The current approach of syncing the spec from registries is as follows:
+ 1. Delete all the specs currently in the datastore
+ 2. Fetch new specs
+ 3. Add all the fetched specs to the datastore
+ 
+This leads to a bunch of bugs [Bug 1583495](https://bugzilla.redhat.com/show_bug.cgi?id=1583495), [Bug 1577810](https://bugzilla.redhat.com/show_bug.cgi?id=1583495), [Bug 1586345](https://bugzilla.redhat.com/show_bug.cgi?id=1586345)
```

Comment:
I think these would be better added as a list with a small summary of the problem.

Reply:
yes I can add a summary describing the problem

Reply:
I think this one needs to be updated `[Bug 1577810](https://bugzilla.redhat.com/show_bug.cgi?id=1583495)`